framework/src/base/MeshGeneratorSystem.C

```diff
@@ -325,9 +329,60 @@ MeshGeneratorSystem::executeMeshGenerators()
   // Order the generators
   createMeshGeneratorOrder();
 
-  std::map<std::string, std::unique_ptr<MeshBase> *> to_save_in_meshes;
+  // Manage the data driven capability, which needs to be done this late
+  // because folks could add generators via append_mesh_generator
+  const auto & moose_mesh = _app.actionWarehouse().getMesh();
+  std::set<const MeshGenerator *> data_only_generators;
+  if (moose_mesh->parameters().get<bool>("_mesh_generator_mesh") &&
+      moose_mesh->isParamValid(data_driven_generator_param))
+  {
+    if (!hasDataDrivenAllowed())
+      moose_mesh->paramError(
+          data_driven_generator_param,
+          "This application does not support data-driven mesh generation.\n\nThis generation is an "
+          "advanced feature and must be enabled on the application via the '",
+          allow_data_driven_param,
+          "' parameter.");
+
+    mooseAssert(moose_mesh->type() == "MeshGeneratorMesh",
+                "Assumption for mesh type is now invalid");
 
-  // Loop over the MeshGenerators and save all meshes marked to to_save_in_meshes
+    const auto & data_driven_generator_name =
+        moose_mesh->parameters().get<std::string>(data_driven_generator_param);
+    if (!hasMeshGenerator(data_driven_generator_name))
+      moose_mesh->paramError(data_driven_generator_param,
+                             "The data driven generator '",
+                             data_driven_generator_name,
+                             "' does not exist");
+
+    // Make sure all parents support data driven generation, and keep track of them
+    for (const auto & name_generator_pair : _mesh_generators)
+    {
+      const auto & generator = *name_generator_pair.second;
+      if (generator.isChildMeshGenerator(data_driven_generator_name, false))
```

Comment:
It's fairly clear what the `direct` parameter means, but some doxygen for it would still be appreciated